### PR TITLE
Update API.md to reflect microtime keyed errors

### PR DESCRIPTION
**Pull request by yuwenmemon**
@iwiznia @bondydaa @neil-marcellini @Gonals @chiragsalian @MariaHCD please review

Update the API.md docs to reflect the fact that we want errors returned in an array keyed by the microtime.

### CODE REVIEW

**Comment by yuwenmemon** [comment]
I'm actually curious how we want to handle the "other additional properties" case - should those also be keyed under the microtime?

**Review by PauloGasparSv — DISMISSED** [review]
LGTM
Just couldn't find Onyx::getErrorMicroTime on main.

@yuwenmemon feel free to merge, not merging so people can have time to look and discuss your last comment.

**Review by MariaHCD — COMMENTED** [review]
> I'm actually curious how we want to handle the "other additional properties" case - should those also be keyed under the microtime?

I'm thinking it makes sense to associate the additional properties with the same microtime key. For instance, in [UpdatePersonalDetailsForWallet](https://github.com/Expensify/Web-Expensify/pull/34326) the API might return an additional prop `'errorCode' => 'ssnError'` and later we might also return a different errorCode `kbaNeeded`. 

If we didn't have the additional properties keyed under microtime the onyxData would look like:

```php
[
    [
        'onyxMethod' => Onyx::METHOD_MERGE,
        'key' => OnyxKeys::WALLET_ADDITIONAL_DETAILS,
        'value' => [
            'errors' => [
                microtime() => 'We\'re having trouble verifying your SSN. Please enter the full 9 digits of your SSN.',
                microtime() => 'We need to ask you a few more questions to finish validating your identity',
            ],
            'errorCode' => self:: WALLET_ERROR_KBA, // This will indicate to the frontend that the user must be redirected to the AnswerQuestionsForWallet step
            'idNumber' => $idNumber,
            'questions' => $questions,
        ],
    ],
];
```

The `errorCode` would be overwritten by the second error and while this wouldn't matter too much for this command since it's pattern C but if it were pattern B, we wouldn't want that prop to be overwritten, correct?

So perhaps it would be better if the onyxData looked like:

```php
[
    [
        'onyxMethod' => Onyx::METHOD_MERGE,
        'key' => OnyxKeys::WALLET_ADDITIONAL_DETAILS,
        'value' => [
            'errors' => [
               microtime() => [
                    'error' => 'We\'re having trouble verifying your SSN. Please enter the full 9 digits of your SSN.',
                    'errorCode' => 'ssnError', // We'll use this code on the frontend to alter the ssn field to accept the full 9 digit ssn
                ]
               microtime() => [
                    'error' => 'We need to ask you a few more questions to finish validating your identity',
                    'errorCode' => self:: WALLET_ERROR_KBA, // This will indicate to the frontend that the user must be redirected to the AnswerQuestionsForWallet step
                    'idNumber' => $idNumber,
                    'questions' => $questions,
                ]
            ],
        ],
    ],
];
```

Thoughts?

File: contributingGuides/API.md [hunk 1]
```diff
@@ -43,14 +43,16 @@ The data will automatically be sent to the user via Pusher.
 #### WRITE Response Errors
 When there is an error on a WRITE response (`jsonCode!==200`), the error must come back to the client on the HTTPS response. The error is only relevant to the client that made the request and it wouldn't make sense to send it out to all connected clients.
 
-Error messages should be returned and stored as a String under the `error` property. If absolutely needed, additional error properties can be stored under other, more specific fields that sit at the same level as `error`:
+Error messages should be returned and stored as an object under the `errors` property, keyed by a [microtime](https://www.php.net/manual/en/function.microtime.php). If absolutely needed, additional error properties can be stored under other, more specific fields that sit at the same level as `errors`:
```
Comment (neil-marcellini) [inline contributingGuides/API.md:46]: > keyed by a microtime

I think you should mention that it is an int, and update it to use `Onyx::getErrorMicroTime()` instead. We can wait until that PR is merged.

**Review by neil-marcellini — CHANGES_REQUESTED** [review]

**Review by iwiznia — DISMISSED** [review]

**Review by PauloGasparSv — DISMISSED** [review]

**Review by chiragsalian — DISMISSED** [review]

**Comment by yuwenmemon** [comment]
@neil-marcellini updated for your suggestion!

**Review by PauloGasparSv — APPROVED** [review]

**Review by chiragsalian — APPROVED** [review]

**Review by neil-marcellini — APPROVED** [review]
Looks good, thanks 😄

**Pull request merged**